### PR TITLE
Rewrite project using VimeoJS official API and VideoJS' plugin generator

### PR DESCRIPTION
**Pull request by jbourassa**
Followup of #92 

I rewrote the Vimeo Tech using the Vimeo Player API. It's still not battle tested ~and somewhat of a work in progress~ (edit: looks good enough for now!), but I think I have the basic scenarios covered.

@gkatsev Let me know what you think!

### CODE REVIEW

**Comment by davekiss** [comment]
@jbourassa I updated the `package.json` main entry to `dist/videojs-vimeo` and I'm running into an error when using `import 'videojs-vimeo'`

Cannot read property 'getComponent' of undefined

coming from this line: https://github.com/didacte/videojs-vimeo/blob/9f809437f5e79c0d4205df74ad6fdf86e339a09f/dist/videojs-vimeo.js#L2069

Any ideas on why this might be happening?

**Comment by jbourassa** [comment]
@davekiss Hmm, that means that videojs isn't loaded properly. Now that might be an issue with your build tool config, because `import 'videojs-vimeo';` obviously isn't ES5.

Have you tried importing videojs first?

**Comment by davekiss** [comment]
@jbourassa i have, my imports are as shown below:

```
import videojs from 'video.js';
import videojsYouTube from 'videojs-youtube';
import 'videojs-vimeo';
```

**Comment by jbourassa** [comment]
Allright, well, at this point I guess I need a trimmed down example w/ your build tool that I can clone and try.

I'm a bit confused since this is generated from Videojs' generator, but I can have a look.

File: src/Vimeo.js [hunk 1]
```diff
@@ -1,374 +1,226 @@
...
-  if(typeof define === 'function' && define.amd) {
-    define(['video.js'], function(videojs){
-      return (root.Vimeo = factory(videojs));
-    });
-  } else if(typeof module === 'object' && module.exports) {
-    module.exports = (root.Vimeo = factory(require('video.js')));
-  } else {
-    root.Vimeo = factory(root.videojs);
+import videojs from 'video.js';
+import VimeoPlayer from '@vimeo/player';
```
Comment (jbourassa) [inline src/Vimeo.js]: That's very true, I fixed it in the other fork (that we're using in production), but I need to update this one too.

Comment (jbourassa) [inline src/Vimeo.js]: ✅

**Comment by davekiss** [comment]
@jbourassa this seems to work if I instead use `import 'videojs-vimeo/src/vimeo';`

**Comment by davekiss** [comment]
@jbourassa @gkatsev any ideas as to why the `useractive` event doesn't seem to work with this tech?

**Comment by jbourassa** [comment]
>any ideas as to why the useractive event doesn't seem to work with this tech?

Vimeo's player doesn't really expose any of that: https://github.com/vimeo/player.js

Looking at [YouTube's code](https://github.com/videojs/videojs-youtube/blob/master/src/Youtube.js), I dont think it'd work with that tech neither.

**Comment by davekiss** [comment]
@jbourassa I've tested with the YouTube player and it actually does work, so I'm not sure exactly how it hooks in

**Comment by revolunet** [comment]
Thanks for this work. work nicely here. i fixed the linting and added support for options.color here : https://github.com/didacte/videojs-vimeo-fork/pull/1

**Comment by revolunet** [comment]
would be cool to review and eventually publish that branch :) thanks

**Comment by gkatsev** [comment]
Yes, it would. I have it on my TODO, sorry it's taken so long, I should get to it soon.

File: package.json [hunk 2]
```diff
@@ -1,24 +1,107 @@
...
-    "url": "https://github.com/eXon/videojs-vimeo/issues"
+  "browserify": {
+    "transform": [
+      "browserify-versionify"
+    ]
+  },
+  "browserify-shim": {
+    "qunit": "global:QUnit",
+    "sinon": "global:sinon",
+    "video.js": "global:videojs"
```
Comment (varemenos) [inline package.json]: Shimming video.js via browserify-shim is what's causing the `Cannot read property 'getComponent' of undefined` mentioned above.

> Makes define and also module be undefined, in order to fix improperly-authored libraries that need shimming but try anyway to use AMD or CommonJS. For more info read the comment inside this fixture
https://github.com/thlorenz/browserify-shim

**Comment by gkatsev** [comment]
Looking at this and I think the main issue I have is that you can't hide vimeo's control bar but that's a limitation of their API it seems.

**Comment by jbourassa** [comment]
Yeah, you can't really hide the native controls without horrible hacks that have a bunch of downsides. I would advise against it, but Plyr does that.

**Comment by gkatsev** [comment]
Seems like you can do it if you have vimeo pro (though, then you also have direct links to the video as well).

Seems like plyr is using the "stretch the iframe out and then crop everything but the video" trick which only works on desktops. Not sure it's worth doing.

**Comment by varemenos** [comment]
> Seems like you can do it if you have vimeo pro (though, then you also have direct links to the video as well).

Only the video publisher can enable it though

**Comment by gkatsev** [comment]
I'm not sure what the vimeo TOS is but it's a common technique that's outlines in several places.

Anyway, I'm going to merge this in now and I'll just make tweaks on top of this.

**Comment by varemenos** [comment]
I've made a few changes here to fit our needs: https://github.com/goodvidio/videojs-vimeo-fork/tree/gv_rewrite It's based on this PR's code.
Let me know if you want any of these changes applied in the upstream repo.

**Comment by gkatsev** [comment]
Yes, please submit PRs with fixes. Highly appreciated.
Now that we've got this PR in, it'll be a lot easier to keep up with maintenance.

**Comment by varemenos** [comment]
I'll try to find some free time during the weekend and check what makes sense to be merged back.

**Comment by gkatsev** [comment]
Awesome, thanks @varemenos.

**Pull request merged**